node/Cargo.toml

```diff
@@ -10,8 +10,10 @@ celestia-types = { workspace = true }
 tendermint-proto = { workspace = true }
 
 async-trait = "0.1.73"
+clap = { version = "4.4.4", features = ["derive"] }
```

Comment:
not sure if I'm happy with clap in a node unconditionally? wdyt

Reply:
True, especially since it's just a single `enum Network`. I'm thinking about duplicate `Network` dedicated just for arg-parsing in `celestia` with `From` and `Into` to keep the one in `node` pure. Alternatively we could feature gate it behind `native-node` or similar.

Reply:
I second the duplication of this enum, that was my first thought but didn't want to push it right away :)

Reply:
Separated the two, so that `ArgNetwork` is responsible for the argument parsing and passing side, while `Network` in `node` references the network in node's context.